Formula/nginx-unit.rb

```diff
@@ -0,0 +1,267 @@
+    #system "./configure", "php"
+    #system "make", "php"
+    #libunit.install "build/php.unit.so"
+
+    system "./configure", "python"
+    system "make", "python"
+    libunit.install "build/python.unit.so"
+
+    system "./configure", "ruby"
+    system "make", "ruby"
+    libunit.install "build/ruby.unit.so"
+  end
+
+  def caveats
+    <<~EOS
+      Once running, you can control unit using the control socket, for example using curl:
+        curl --unix-socket #{var}/run/unit.control.sock http://localhost/
+
+      Modules are available for system perl, python and ruby. The system php doesn't support nginx-unit. You can install additional modules into:
+        #{HOMEBREW_PREFIX/"lib/unit"}
+    EOS
+  end
+
+  plist_options :manual => "unitd"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>KeepAlive</key>
+        <false/>
+        <key>ProgramArguments</key>
+        <array>
+            <string>#{opt_bin}/unitd</string>
+            <string>--no-daemon</string>
+        </array>
+        <key>WorkingDirectory</key>
+        <string>#{HOMEBREW_PREFIX}</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/unitd", "--version"
+
+    require "json"
+    require "net/http"
+    require "socket"
+    require "timeout"
+
+    logfile = var/"log/unit.log"
+    logfile.unlink if logfile.exist?
+
+    pidfile = var/"run/unit.pid"
+    pidfile.unlink if pidfile.exist?
+
+    begin
+      # We need to install rack for the ruby module test to work, so set a
+      # temporary gems path before booting unitd. (unitd boots and functions
+      # fine without, but fails to start an app until rack is installed.)
+      rubydir = testpath/"ruby"
+      rubygems = rubydir/"gems"
+      rubygems.mkpath
+      ENV["GEM_PATH"] = rubygems.to_s
+      system "gem", "install", "rack", "--install-dir=#{rubygems}", "--verbose"
+
+      pid = spawn "#{bin}/unitd", "--no-daemon"
+
+      timeout(5) { sleep 0.1 until pidfile.exist? }
+
+      assert_equal pid, pidfile.read.chomp.to_i
+
+      timeout(5) { sleep 0.1 until File.exist? var/"run/unit.control.sock" }
+
+      # Unit uses a unix control socket. Net::HTTP to a socket is hard. Sorry.
+      socket = Net::BufferedIO.new(UNIXSocket.new(var/"run/unit.control.sock"))
+      request = Net::HTTP::Get.new("/")
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) { }
+
+      assert_equal("200", response.code)
+      expected_body = {"listeners" => {}, "applications" => {}}
+      assert_equal expected_body, JSON.parse(response.body)
+
+      assert logfile.size > 0
```

Comment:
You'd want some kind of failure message here or it'll potentially blow up in a way that'll be completely and utterly unhelpful to anyone.

Reply:
```
````

seems to fail nicely:

```